### PR TITLE
chore: refactor to use layout service for onPageLoad actions calculations in blocks API

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Refactor the consolidated API for importing blocks

Fixes 
## Automation

/ok-to-test tags="@tag.Templates"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9690978421>
> Commit: ce3884a3db7b99cd36b0d600b43120d8eafb61fb
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9690978421&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Templates`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Refactor**
  - Improved code structure by extending interface and class compatibility for `ConsolidatedAPIService` and `ConsolidatedAPIServiceImpl`.

- **New Features**
  - Enhanced layout management by importing `UpdateLayoutService` and updating related logic to improve page load actions.

### CODE REVIEW

**Comment by abhvsn** [comment]
Adding @subrata71 as I don't have much context here.

**Review by subrata71 — CHANGES_REQUESTED** [review]
Missing tests

**Comment by AnaghHegde** [comment]
@subrata71 Tests are already covered, this is just a refactor to include the newly created entities in the block API response.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCE.java [hunk 1]
```diff
@@ -18,4 +23,7 @@ Mono<Integer> updateMultipleLayouts(
     JSONObject unescapeMongoSpecialCharacters(Layout layout);
 
     Mono<String> updatePageLayoutsByPageId(String pageId);
+
+    Mono<List<Set<DslExecutableDTO>>> getOnPageLoadActions(
```
[the remark this answers is not in the document]
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCE.java:27]: The method is not tailored to building blocks alone. Tis can be resused for any scenarios where you want to extract the onPageLoadActions for a given DSL. This is currently useful for blocks only. The method accepts the stringified dsl.

[the remark this answers is not in the document]
Comment (subrata71) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCE.java:27]: In that case, can you move this method to `OnLoadExecutablesUtilCE`?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCEImpl.java [hunk 2]
```diff
@@ -304,6 +304,67 @@ public Mono<String> updatePageLayoutsByPageId(String pageId) {
...
+                    .then(Mono.error(t));
+        }
+
+        layout.setWidgetNames(widgetNames);
+
+        if (!escapedWidgetNames.isEmpty()) {
+            layout.setMongoEscapedWidgetNames(escapedWidgetNames);
+        }
+        Set<ExecutableDependencyEdge> edges = new HashSet<>();
+        Set<String> executablesUsedInDSL = new HashSet<>();
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCEImpl.java:337]: At what point do we expect the executables to be populated?

Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/layouts/UpdateLayoutServiceCEImpl.java:337]: This method is responsible for populating the data. https://github.com/appsmithorg/appsmith/pull/34269/files#diff-4b01ebbfa24808881d82e433b30e73eaa016982635447979da129eaf1269916bR345

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**